assets/animations.js

```diff
@@ -0,0 +1,25 @@
+function onIntersection(elements, observer) {
+  elements.forEach((element, index) => {
+    if (element.isIntersecting) {
+      const elementTarget = element.target;
+      elementTarget.classList.add('animation__trigger--active');
+      elementTarget.setAttribute('style', `--animation-order: ${index};`);
```

Comment:
What do you see this being used for? It might be helpful for cascading animations although those will depend on the screen size and layout.

Reply:
This is for cascading animations, yes. And it's to replace [all this CSS](https://github.com/Shopify/dawn/blob/ebc1032459cabfeadbccc6fd3a9a43ae2a65bea5/assets/base.css#L3008-L3077) in the prototype.

Reply:
Unfortunately you'll need that CSS I think to accommodate different breakpoints. On product grids, for example, you will cascade 4 items at a time at desktop size, but 1 or 2 at mobile size.

Anyhow I suggest leaving this for a subsequent PR and focus on the intersection observer for now.